components/compliance-service/reporting/util/csv.go

```diff
@@ -82,3 +83,10 @@ func ReportToCSV(report *reportingapi.Report) (string, error) {
 	}
 	return content, nil
 }
+
+func maxCharLimit(item string) string {
+	if utf8.RuneCountInString(item) >= 32000 {
+		return fmt.Sprintf("character limit exceeds 32,000. truncating message: %s", item[0:31000])
```

Comment:
Do we care about non-ASCII data here?

The slice indexing here assuming that all of your characters are 1-byte characters (i.e. restricted to the ASCII range of UTF-8).

I think you can do something like:
````golang
string([]rune(item)[0:310000])
````

 but I'm unsure on the runtime cost of the conversion.  Perhaps something like this might be cheaper, we'd have to test:

```golang
func safeTruncate(s string, size int) string {
	truncated := s[0:size+1]
	for {
		r, _ := utf8.DecodeRune([]byte{truncated[len(truncated)-1]})
		if r == utf8.RuneError {
		   truncated = truncated[0:len(truncated) - 1]
		} else {
		   break
		}

	} 
	return truncated
}
```

Here you can see an example of the malformed character you can get by indexing into a string that might have utf8 data:

https://play.golang.org/p/PRkuh4EeJ4R

I'm also wonder if, since this is a very long string, we might want to play with how we build the truncation message.  That is, I wonder if strings.Builder or some other method might be more efficient than Sprintf with a string that we know is 31k.

Reply:
interesting, thank you!! i'll play around with those two solutions